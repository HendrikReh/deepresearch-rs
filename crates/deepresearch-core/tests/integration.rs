use deepresearch_core::run_research_session;

#[tokio::test]
async fn critic_verdict_is_non_empty() {
    let summary = run_research_session("Assess lithium battery market drivers 2024")
        .await
        .expect("workflow should succeed");

    assert!(
        summary.contains("Sources:"),
        "summary should list sources: {summary}"
    );

    assert!(
        summary.contains("Analysis passes"),
        "expected critic verdict wording: {summary}"
    );
}
