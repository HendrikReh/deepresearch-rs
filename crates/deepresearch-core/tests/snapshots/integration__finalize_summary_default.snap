---
source: crates/deepresearch-core/tests/integration.rs
expression: summary
---
Analysis passes automated checks

Summary:
Top insights: Automated placeholder insight. Additional manual review recommended.. Confidence supported by 1 sources.

Key Insight: Automated placeholder insight. Additional manual review recommended.
Confidence: High
Sources:
  1. stub://memory

Fact-Check Confidence: 1.00
Verified Sources:
  1. stub://memory
